webapp/src/js/controllers/messages-content.js

```diff
@@ -115,9 +115,35 @@ angular.module('inboxControllers').controller('MessagesContentCtrl',
         getContactable(id, type),
         MessageContacts.conversation(id)
       ])
+        .catch(function(err) {
+          if (err.code === 404) {
+            return $q.all([Promise.resolve(null), MessageContacts.conversation(id)]);
+          }
+
+          throw err;
+        })
         .then(function(results) {
           var contactModel = results[0];
```

Comment:
While you're here feel free to change these `var` into `const` and `let`.

Here especially it's actually quite useful: it's letting us know that `contactModel` may change, which would lead us to look further into the code and notice that code path.